apps/chat/src/store/models/models.reducers.ts

```diff
@@ -376,13 +436,39 @@ const selectInitialized = createSelector(
   (state) => state.initialized,
 );
 
+const selectCustomModels = createSelector([rootSelector], (state) => {
+  return state.models.filter((model) => model.reference !== model.id);
+});
+
+const selectSharedWithMeModels = createSelector(
+  [selectCustomModels],
+  (customModels) => {
+    return customModels.filter((model) => model.sharedWithMe);
+  },
+);
+
+const selectSharedWriteModels = createSelector(
+  [selectCustomModels],
+  (customModels) => {
+    return customModels.filter((model) => canWriteSharedWithMe(model));
+  },
+);
+
+const selectIsSharedWithMeModelById = createSelector(
+  [selectModelsMap, (_state, modelId: string | undefined) => modelId],
+  (modelsMap, modelId) => {
+    return modelId ? modelsMap[modelId]?.sharedWithMe : modelId;
```

Comment:
Why do you return `modelId` here if check `sharedWithMe`?
```suggestion
    return modelId ? modelsMap[modelId]?.sharedWithMe : false;
```

Reply:
Thank you for the suggestion.
The logic has been changed and this selector has been removed as redundant.